agent/rpc/middleware/interceptors.go

```diff
@@ -52,9 +52,8 @@ func (r *RequestRecorder) Record(requestName string, rpcType string, start time.
 		{Name: "rpc_type", Value: rpcType},
 	}
 
-	// TODO(FFMMM): it'd be neat if we could actually pass the elapsed observed above
-	r.recorderFunc(metricRPCRequest, start, labels)
-
+	// math.MaxInt64 < math.MaxFloat32 is true so we should be good!
+	r.recorderFunc(metricRPCRequest, float32(elapsed), labels)
```

Comment:
```suggestion
	r.recorderFunc(metricRPCRequest, float32(elapsed.Milliseconds()), labels)
```

You are right about the maxes. However most of our timing related metrics are in MS so we should convert the nanosecond duration to milliseconds here. go-metrics timings force ms precision.